guardian/prosemirror-typerighter
# Add error messaging

## What does this change?


At the moment, when we make requests to the Typerighter API, it's not clear if something has gone wrong or not.

In this PR we add an error state to the UI that reflects any problems, meeting these requirements:

- The error state should describe what's wrong in as clear a way as possible to the user, who will not be a developer.
- The error should clearly be 'owned' by Typerighter, not the parent tool. This probably means colocating it in existing UI.
- The error should be dismissable.

## How to test


To emulate a partial failure on your local environment, you can go to the Typerighter `ApiController.scala` and use `scala.util.random` to reject arbitrary requests.

To emulate a complete failure, block the Typerighter domain requests.

Click 'Check whole document' and the error message should appear below.

## How can we measure success?

A user is able to tell when something has gone wrong and is informed of what actions to take to rectify the situation.

## Images

![TyperighterError](https://user-images.githubusercontent.com/4633246/89199844-17933700-d5a7-11ea-86e9-17879bb829d3.gif)
